diff --git a/src/socket.c b/src/socket.c
index 7ba231a..2114df8 100644
--- a/src/socket.c
+++ b/src/socket.c
@@ -831,7 +831,7 @@ void ReceiveMsg(void)
        case MSG_ATTACH:
                if (CreateTempDisplay(&m, recvfd, win))
                        break;
-               AskPassword(&m);
+               FinishAttach(&m);
                break;
        case MSG_ERROR:
                {
@@ -850,7 +850,7 @@ void ReceiveMsg(void)
        case MSG_POW_DETACH:
                if (CreateTempDisplay(&m, recvfd, NULL))
                        break;
-               AskPassword(&m);
+               FinishDetach(&m);
                break;
        case MSG_QUERY:
                {
